File: DrivingLicense.js
const UniqueItem = require("./UniqueItem");

const prompt = require("prompt-sync")();
class DrivingLicense extends UniqueItem {
  constructor(lastName, firstName, expirationDate) {
    super("CreditCard");
    this.lastName = lastName;
    this.firstName = firstName;
    this.expirationDate = expirationDate;
  }

  static createDrivingLicense() {
    const lastName = prompt("Entrez le nom sur le permis de conduire : ");
    const firstName = prompt("Entrez le prénom sur le permis de conduire : ");
    const expirationDate = prompt(
      "Entrez la date d'expiration du permis de conduire (MM/YYYY) : ",
    );

    // Vous pouvez ajouter une logique de validation pour la date d'expiration si nécessaire

    return new DrivingLicense(lastName, firstName, expirationDate);
  }

  displayInfo() {
    console.log("Informations du permis de conduire :");
    console.log(`Nom : ${this.lastName}`);
    console.log(`Prénom : ${this.firstName}`);
    console.log(`Date d'expiration : ${this.expirationDate}`);
  }
}

module.exports = DrivingLicense;

File: DrivingLicense.py
from UniqueItem import UniqueItem
from UtilsMethod import UtilsMethod


class DrivingLicense(UniqueItem):
    def __init__(self, last_name, first_name, expiration_date, name):
        super().__init__(name)
        self.last_name = last_name
        self.first_name = first_name
        self.expiration_date = expiration_date

    @classmethod
    def from_user_input():
        print("Enter last name: ")
        last_name = input()
        print("Enter first name: ")
        first_name = input()
        print("Enter expiration date: ")
        expiration_date = UtilsMethod.ask_date(input)

        # Assuming 'name' is the name of the driving license
        return DrivingLicense(last_name, first_name, expiration_date, "DrivingLicense")

    def __str__(self):
        return (
            f"DrivingLicense{{lastName='{self.last_name}', firstName='{self.first_name}', "
            f"expirationDate={self.expiration_date}}}"
        )
